docs/design-notes/reverse-reads.md

```diff
@@ -0,0 +1,121 @@
+# Reverse reads
+
+A read is called reverse when it reads with reverse clustering order
+(compared to that of the schema). Example:
+
+    CREATE TABLE mytable (
+        pk int,
+        ck int,
+        s int STATIC,
+        v int,
+        PRIMARY KEY (pk, ck)
+    ) WITH
+        CLUSTERING ORDER BY (ck ASC);
+
+    # Forward read (using table's native order)
+    SELECT * FROM mytable WHERE pk = 1;
+    # Explicit forward order
+    SELECT * FROM mytable WHERE pk = 1 ORDER BY ck ASC;
+
+    # Reverse read
+    SELECT * FROM mytable WHERE pk = 1 ORDER BY ck DESC;
+
+If the table's native clustering order is DESC, then a read with ASC
+order is considered reverse.
+
+## Legacy format
+
+The legacy format is how scylla handled reverse queries internally. We
+are in the process of migrating to the native reverse format, but for
+now coordinator-side code still uses the legacy format.
+
+### Request
+
+The `query::partition_slice::options::reversed` flag is set.
+Clustering ranges in both `query::partition_slice::_row_ranges` and
+`query::specific_ranges::_ranges`
+(`query::partition_slice::_specific_ranges`) are half-reversed: they
+are ordered in reverse, their end bound used as start bound and
+vice-versa.
```

Comment:
This contradicts the example where the start/end bounds have not been reversed, just the vector.

(from the query order point of view, the start/end bounds appear reversed, because they are still in table order)

Reply:
I doesn't, I'm talking about how the bounds are *used*. They are left as-is, as they are in a forward slice.
I can rephrase this to make this more clear.

Reply:
Clarified in v9